packages/core/src/App/Containers/AccountSignupModal/account-signup-modal.jsx

```diff
@@ -90,11 +116,16 @@ const AccountSignup = ({ enableApp, isModalVisible, clients_country, onSignup, r
     };
 
     const validateSignupPassthrough = values => validateSignup(values, residence_list);
+
     const onSignupPassthrough = values => {
         const index_of_selection = residence_list.findIndex(
             item => item.text.toLowerCase() === values.residence.toLowerCase()
         );
 
+        if (!is_eu_resident) {
+            values.email_consent = true;
```

Comment:
It’s not a best practice to change an argument of a function.

Reply:
I am not sure how I can resolve this better. Somehow I have to intercept the value before it gets send to the websocket, and make sure that the user always gets 1 (true) if he/she is non-EU.